### DISCUSSION TITLE
SharePoint PnP Search Box Not Returning Results

### DISCUSSION THREAD

**Discussion by lokesh6116** [microsoft-search/pnp-modern-search#3862] [Q&A] [answered]
Description: 
When entering the search term that is visible in the search results on SharePoint PnP Search box, no results are displayed in the search box dropdown. This occurs consistently across different browsers( chrome, edge) and in our SharePoint Online environment.

Steps to Reproduce:

1. Navigate to the SharePoint PnP site.
2. Enter the search term " example in the search box.
3. Expected: Dropdown should display results including items with " example".
4. Actual: Dropdown shows no results  despite " example being visible in the search results page.

Environment:
-SharePoint Online
-PnP Version: 4.12.2.0
- Browser:

Additional Context:
- I can see JavaScript errors on the console.
- We recently updated to the latest version of PnP, but the issue persists.

Screenshots:

Logs/Errors:

**Comment by kasperbolarsen** [comment]
I have to assume that you have enabled query suggestions on the search box configuration? Can you share the settings?

**Comment by lokesh6116** [comment]
@kasperbolarsen  Thanks for responding

I have turned off query suggestions still no results are displaying

Page 1 : 

Search box settings
Placeholder text to display in the search box : Empty
Send the query to a new page : Off
Reset query on clear : Off

Page 2 : 

Query suggestions
Enable query suggestions : On
Suggestion providers
Number of suggestions to show per group : 10

Page 3 : 

Available connections

Use dynamic data source as default input : On
Connect to source :  PnP - Search Results's

PnP - Search Results's properties : Search results data

Search results data - cfcb0a9f-9881-4331-8866-a484f13612f8's properties : Selected Items

**Comment by lokesh6116** [comment]
Version : 4.12.2.0

**Comment by kasperbolarsen** [comment]
add query suggestions are enabled in the SharePoint Admin center?
![image](https://github.com/user-attachments/assets/35174a88-056c-43db-98b9-1bd66dd98aae)

**Comment by lokesh6116** [comment]
Yes, its enabled.

**Comment by kasperbolarsen** [comment]
Just to be sure we are talking about the same thing: are you aiming for getting the static search suggestions working or have you created your own like documented here https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_suggestions_provider/

**Comment by lokesh6116** [comment]
You are right, aiming to get static search suggestions which has to be from the PnP search results webpart.

**Comment by kasperbolarsen** [comment, marked as answer]
Static search suggestions are static as per https://learn.microsoft.com/en-us/sharepoint/search/manage-query-suggestions
And will not be related to the results .

**Reply by wobba** [reply]
This is the correct answer. Suggestions will return suggestions manually uploaded. Any "learned" suggestion will likely not happen because of details too complex - and a feature likely going away from classic anyways.

**Comment by lokesh6116** [comment]
Does that mean we can't search from the search results data?

**Comment by kasperbolarsen** [comment]
We do support Search suggestions extentions as per https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_suggestions_provider/

**Comment by lokesh6116** [comment]
We have the search results and when I search for a term from the search results in the search box, it doesn't show the results is this an expected behavior?

**Comment by kasperbolarsen** [comment]
It depends,  if your term value is "India" and you just search for "India" in the search and your search query in the Results web part is {searchterm} , then yes, that is expected.
The reason is that the RefinableStrings are not queryable. You can test this by changing the search text to "RefinableStringXX:"India" 
Replace XX with the matching RefinableString

**Comment by lokesh6116** [comment]
I dont see any option to connect search box to search results. I am not able to find the option Connect to a search box or to set up dynamic data source section in search results web part settings.

**Reply by kasperbolarsen** [reply]
See "page" 3 in the Search Results web part, assuming you are using a version 4.x

**Reply by lokesh6116** [reply]
I am using 4.12.2.0 

I see the below four options under PnP-Search Results webpart - Available connections page 3 of 4
Use input query text Off
Connect to a filters Web Part Off
Connect to a verticals Web Part Off
Connect to a search results Web Part Off

**Reply by kasperbolarsen** [reply]
Set Use input query text to On, and you will see all of the options 
 ![PnP-modern-search-filters-connect-to-search-results-2-1024x605.png](https://github.com/user-attachments/assets/831729a0-49f4-4050-a687-efa188980809)

**Reply by lokesh6116** [reply]
Thanks. That worked
